core/modules/update.cc

```diff
@@ -34,48 +34,37 @@ void Update::ProcessBatch(bess::PacketBatch *batch) {
 }
 
 CommandResponse Update::CommandAdd(const bess::pb::UpdateArg &arg) {
-  int curr = num_fields_;
+  size_t curr = num_fields_;
 
-  if (curr + arg.fields_size() > UPDATE_MAX_FIELDS) {
-    return CommandFailure(EINVAL, "max %d variables can be specified",
-                          UPDATE_MAX_FIELDS);
+  if (curr + arg.fields_size() > kMaxFields) {
+    return CommandFailure(EINVAL, "max %zu variables can be specified",
+                          kMaxFields);
   }
 
   for (int i = 0; i < arg.fields_size(); i++) {
     const auto &field = arg.fields(i);
 
-    uint8_t size;
-    int16_t offset;
-    uint64_t mask;
-    uint64_t value;
-
-    offset = field.offset();
-
-    size = field.size();
+    size_t size = field.size();
     if (size < 1 || size > 8) {
       return CommandFailure(EINVAL, "'size' must be 1-8");
     }
 
-    if (!bess::utils::uint64_to_bin(&value, field.value(), size,
-                                    bess::utils::is_be_system())) {
-      return CommandFailure(
-          EINVAL, "'value' field has not a correct %d-byte value", size);
-    }
-
-    if (offset < 0) {
-      return CommandFailure(EINVAL, "too small 'offset'");
+    if (field.offset() + size > SNBUF_DATA) {
+      return CommandFailure(EINVAL, "too large 'offset'");
     }
 
-    offset -= (8 - size);
-    mask = (1ull << ((8 - size) * 8)) - 1;
+    be64_t value(field.value() << ((8 - size) * 8));
```

Comment:
`samples/update.bess` covers both `Update` and `RandomUpdate` (which has been updated similarly).